Source/JavaScriptCore/runtime/VMTraps.cpp

```diff
@@ -362,6 +363,11 @@ void VMTraps::fireTrap(VMTraps::Event event)
         m_condition->notifyAll(locker);
     }
 #endif
+
+    if (event == NeedTermination) {
+        if (vm().hasSyncWaiter())
+            vm().syncWaiter()->condition().notifyOne();
+    }
```

Comment:
Instead of doing this here, let's do it inside VMTraps::SignalSender::work, which can do this repeatedly until VM gets terminated. And also, we should do it after `sendMessage` gets sent. So you can just do it before the shutting down check.

Reply:
Discussed. Add the notification invocation to both here and `sendMessage`.